### DISCUSSION TITLE
[RFC] Ideas for Multimedia Collections

### DISCUSSION THREAD

**Discussion by AdultSun** [stashapp/stash#2127] [Ideas]
This is a summary of a recent conversation in the #requests channel on Discord. The discussion starts around the following post, [linked here](https://discord.com/channels/559159668438728723/559399634318131210/920517283359629343). It's a long post, so I gave each section a heading. Hopefully that makes it easier to read for everybody and for the devs to reference in the future.

# Overview
### Including Videos in Galleries
An increasingly frequent request is to allow for scenes to be included in galleries. The primary use case for this is to cover sets downloaded from OnlyFans and similar sites, which often include short clips mixed in with the photo gallery. With the current feature set, the best Stash can do is link to these clips as separate scenes. This keeps the video clips isolated from the rest of the gallery, and the association has to be manually assigned. The current system is workable, but inelegant and inconvenient for this situation.

### Video Playlists vs. Galleries
A second frequent request is the ability to save video playlists. If galleries can also include videos though, then this creates a lot of crossover between the two concepts -- galleries and playlists -- since a gallery could also just hold a collection of videos. (This is also similar to the existing Movie system, though this is likely a unique enough object-type to differentiate it from the other two.) It may make more sense to make Gallery objects more flexible in their usage, to also cover the functions of a video playlist.

### Changing Name?
Another concern is that expanding the function of Galleries to also include videos would suddenly make the term "Gallery" a bad fit for the feature because it implies a collection of images only. Several commenters like the more general term "Collection" as a better fit instead, though "Gallery" may still fit well enough to leave as-is. (For the sake of this post, I will be using "Collection" to refer to theoretical groupings of multiple media types and "Gallery" for the existing system of images only.)

### Adding to Collections
The current system of adding items to a gallery is also insufficient when expanding functionality to multimedia collections. The two current methods are 1) automatic creation during the Scan task and 2) the use of an "Add" tab that changes the grid view in a gallery's page to allow for filtering, sorting, selecting, and adding images from within Stash. The first relies on folders and zip archives filled with images. This could become problematic when other media types can be considered for inclusion in Collections. The second is unwieldy as it currently exists. It could become more difficult if you need to sort through multiple different media types as well.

### Static vs. Dynamic
A final consideration is whether collections should remain as static lists or if they could be automatically populated. This would be similar to auto-playlist features in programs like Plex. Rules could be determined for an individual collection (includes this tag, duration is < 10 min., etc.) and items would be found and added automatically on a continual basis. This idea may be out of scope for this feature and felt like a loosely related tangent pursued by... me. (Sorry about that.)

# Planning
### What Should be Included in a Collection
I feel the general consensus of this discussion is that Galleries should allow for the inclusion of different types of media and possibly be renamed Collections. Existing object types that could be included are Scenes, Images, and Markers. (Markers may need their own suite of improvements before including them in Collections, however.) Movies and other Galleries/Collections could theoretically be included as well, though it's unclear how that would work. Audio files, comics, and ebooks are media types that have been requested for support within Stash that could be considered for inclusion too.

### Custom Ordering
The existing Gallery system also has no way of manually ordering the images inside of it, other than renaming everything. If an expanded Collection system is meant to cover the function of video playlists as well, then the ability to customize the order of the collection will be necessary.

### Multimedia Playback
The ability to "play" an ordered queue of various media types will also require an **expansion of the lightbox system**, currently only available for images in Wall view. @WithoutPants mentioned he had already wanted to expand the lightbox system to also play videos and eventually audio anyway.

### Adding to Collections
Lastly, we would need more intuitive methods of adding content to Collections than is currently possible with Galleries. The easiest solution would be an "Add to Collection" option in the overflow menus. This would need to be possible from within the detail pages of individual objects as well as from the mass edit views of the media category pages. There should also be an option to save the current video queue as a permanent collection or add it to an existing one. The existing option of creating galleries/collections from folders and zip archives with images inside could remain as is. If this becomes problematic for users dealing with multimedia collections, it can be addressed at that time.

# Request for Comment
Everyone who took part in the discussion will be pinged on Discord with a link to this post. Any questions, additions, corrections, or clarifications are encouraged from anyone reading this. I can repost comments from Discord on here for anyone without a GitHub account, with or without your username.

**Comment by ChilledSlim** [comment]
I will add in, that sometimes adding a lot of items to Stash, like a siterip as suggested, can clutter scenes considerably. A siterip may include hundreds of 7-30 second videos as well as images. Having an option to hide the items (from scenes and gallery) when loaded into a collection would be a way to deal with this.

another addition is to do this via filters (though number of galleries it is a part of is not a current image or scene filter) but having an option to automatically add one (and they’d not always have to have a filter on) would be very helpful. I don’t want hundreds of short clips, but a gallery of short clips I can call on specifically.
